### ISSUE TITLE
TessPdfRenderer not working with jpg files

### ISSUE THREAD

**Issue by PabloRodrizHp** [adaptech-cz/Tesseract4Android#31]
Hi.

Having used the [alexcohn/tess-two](https://github.com/alexcohn/tess-two) repository, the TessPdfRenderer works with jpg files, but when using the latest version of this library, it doesn't. The generated PDF apparently has the text in it, but not the image, and the workaround is to create a PNG out of the JPG file, which in some situations adds up to 2 seconds of processing. This situation is quite similar to the old [issue ](https://github.com/rmtheis/tess-two/issues/122)(from 2015) found in the original [rmtheis](https://github.com/rmtheis/tess-two/tree/master) repository. 

Here is the code that works with `'com.rmtheis:tess-two:9.1.0'` but not this library:

```
TessBaseAPI mTess = new TessBaseAPI();
mTess.setDebug(true);
mTess.init(DATA_PATH, lang);

String pdfOutput = Environment.getExternalStorageDirectory().toString() + "/Download/ocrOutput";
String jpegInput = Environment.getExternalStorageDirectory().toString() + "/Download/jpegInput.jpg";
TessPdfRenderer renderer = new TessPdfRenderer(mTess, pdfOutput);
    
mTess.beginDocument(renderer);

File file = new File(jpegInput);
Pix pix = ReadFile.readFile(file);

boolean addedPageOne = mTess.addPageToDocument(pix, file.getAbsolutePath(), renderer);
Log.e(TAG, "convertImageToSearchablePdf: addedPageOne: " + addedPageOne);

boolean endDocument = mTess.endDocument(renderer);
Log.e(TAG, "convertImageToSearchablePdf: endDocument: " + endDocument );

renderer.recycle();
pix.recycle();
```

Am I missing something in my code, so it works with the other library and not this one?

**Comment by Robyer** [comment]
Hi, I have made no intentional changes to the TessPdfRenderer, so I would expect it should work same as in tess-two library. Based on your link it probably has something to do with missing support for libjpeg, maybe I missed some parameter when compiling Leptonica, to enable the libjpeg support.

If that is the case, I expect that line `Pix pix = ReadFile.readFile(file);` can't read the file and `pix` is then null, is that correct?

**Comment by PabloRodrizHp** [comment]
Hi.

No is not, `pix` is not null. 
And the generated PDF file, when opened appears with a "There was an error processing a page. There was a problem reading this document(18)"
When I dismiss this error, and I can "see" the document, there is no image, but there is a white rectangle with the same size of the original image, and I can even select text where the original image has text:

![image](https://user-images.githubusercontent.com/69145153/109143549-76491980-7760-11eb-972f-ce0fc62657a6.png)
(There should be an ID there)
Trying to copy the selected text, gives the same error as when opening the file.


Also I found something interesting, (probably because I dont know how this works or how the library is compiled).

When I check the code of the classes `TessPdfRenderer`,` ReadFile`, and `Pix`, in the library `cz.adaptech.android:tesseract4android:2.1.0`  they are loading the library "png":

![1-TessPdfRenderer-Library_2](https://user-images.githubusercontent.com/69145153/109144884-0e93ce00-7762-11eb-810a-5667e1f91802.PNG)
![1-Pix-Library_2](https://user-images.githubusercontent.com/69145153/109144887-0f2c6480-7762-11eb-8f17-39a6cf55e651.PNG)
![1-ReadFile-Library_2](https://user-images.githubusercontent.com/69145153/109144888-0f2c6480-7762-11eb-88f6-c6b8a23d9128.PNG)

but in the library `com.rmtheis:tess-two:9.1.0` is "pngt":

![image](https://user-images.githubusercontent.com/69145153/109145116-561a5a00-7762-11eb-8b5f-d1f4de35b476.png)
![image](https://user-images.githubusercontent.com/69145153/109145140-603c5880-7762-11eb-9918-7aef23e215f9.png)
![image](https://user-images.githubusercontent.com/69145153/109145172-6cc0b100-7762-11eb-8415-b535c1af9044.png)

Could this mean that is a different library vesion, and that could cause the malfunction? Because actually `com.rmtheis:tess-two:9.1.0` is using libjpeg 9b, and `cz.adaptech.android:tesseract4android:2.1.0` is using libjpeg 9c.

Cheers.

**Comment by Robyer** [comment]
@PabloRodrizHp I see. In that case you can try:
- 5.0.0 tesseract branch (in case there is some error in Tesseract) - https://github.com/adaptech-cz/Tesseract4Android/tree/tesseract-5.0.0
- or use that older version of libjpeg library (9b) - basically just swap the libjpeg source files in `/tesseract4android/src/main/cpp/libjpeg/src` directory and recompile whole library.

Btw the difference between jpgt / jpeg names is not relevant, I just wanted to use more explicit names for the libraries.

**Comment by PabloRodrizHp** [comment]
Sorry @Robyer . Maybe I missed to explain that I am not compiling the library, I am using the dependency. 

So I guess that my approach for now, will be to try 5.0.0 tesseract branch, which dependency is `cz.adaptech.android:tesseract4android:2.2.0-dev` right?

**Comment by Robyer** [comment]
@PabloRodrizHp 
> Sorry @Robyer . Maybe I missed to explain that I am not compiling the library, I am using the dependency.

I see. In that case you are not really using _alexcohn's tess-two_, because he doesn't provide compiled library - the README is copied from the original _rmtheis' tess-two_ so the dependency line that you copy to your build.gradle is really the original tess-two library using Tesseract 3.x. I just wanted to clarify this. I couldn't even make alexcohn's library work when I compiled manually his latest code.

Anyway, I've reproduced the problem and found the issue. It is caused by change in Tesseract code itself, but it's easy to work-around. **I have just commited the fix to master branch. You just need to compile this library yourself for now,** as I don't have time to release the new version yet.
